src/com/google/enterprise/adaptor/ContentTransformFactory.java

```diff
@@ -0,0 +1,75 @@
+package com.google.enterprise.adaptor;
+
+import com.google.common.base.Strings;
+
+import java.io.OutputStream;
+import java.lang.reflect.Constructor;
+import java.util.List;
+import java.util.Map;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+/**
+ * The content transform factory holds all document content transformers
+ * and puts them in series connection.
+ *
+ * @author Dominik Weidenfeld (dominik.weidenfeld@twt.de)
+ */
+public class ContentTransformFactory {
+
+  private static final Logger LOG =
+      Logger.getLogger(ContentTransformFactory.class.getName());
+
+  private List<Map<String, String>> contentTransformers;
```

Comment:
I changed the implementation, so that all transformers are instantiated in the constructor.
The other "non-optional" parameters are now passed by setters, that guarantees their correctness.